esp/platform/espcontext.cpp

```diff
@@ -112,13 +117,18 @@ class CEspContext : public CInterface, implements IEspContext
         updateTraceSummaryHeader();
         m_secureContext.setown(secureContext);
         m_SecurityHandler.setSecureContext(secureContext);
+        appendGloballyUniqueId(m_localId);
+        // use localId as globalId unless we receive another
+        m_globalId.set(m_localId);
     }
 
     ~CEspContext()
     {
         flushTraceSummary();
+        if (m_txSummaryProfile)
+            m_txSummaryProfile->tailorSummary(this);
```

Comment:
Profile should only be handled if a summary also exists.

Reply:
Adjusted as part of moving ownership of profile from context to txsummary.